bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitCarbonMonoxideSensorImpl.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.io.homekit.internal.accessories;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.items.GenericItem;
+import org.eclipse.smarthome.core.items.ItemRegistry;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.openhab.io.homekit.internal.HomekitAccessoryUpdater;
+import org.openhab.io.homekit.internal.HomekitTaggedItem;
+import org.openhab.io.homekit.internal.battery.BatteryStatus;
+
+import com.beowulfe.hap.HomekitCharacteristicChangeCallback;
+import com.beowulfe.hap.accessories.BatteryStatusAccessory;
+import com.beowulfe.hap.accessories.CarbonMonoxideSensor;
+import com.beowulfe.hap.accessories.properties.CarbonMonoxideDetectedState;
+
+/**
+ *
+ * @author Cody Cutrer - Initial contribution
+ */
+public class HomekitCarbonMonoxideSensorImpl extends AbstractHomekitAccessoryImpl<GenericItem>
+        implements CarbonMonoxideSensor, BatteryStatusAccessory {
+
+    @NonNull
```

Comment:
Annotate the class with `@NonNullByDefault` and use `@Nullable` where applicable. See [Null annotations](https://www.openhab.org/docs/developer/guidelines.html#null-annotations).

Reply:
That would require NullAnnotations for the whole binding.

Reply:
@wborn, I tried to do that. However, I have the following problem and no idea how to solve that.

```
@Override
public CompletableFuture<Boolean> getLightbulbPowerState() {
    final GenericItem item = getItem();
    if (item == null) {
        return CompletableFuture.completedFuture(null);
    }
    OnOffType state = item.getStateAs(OnOffType.class);
    return CompletableFuture.completedFuture(state == OnOffType.ON);
}
```

The method is inherited from an external non-null-annotated library. Eclipse sees two errors: 

l. 2:
```
The return type is incompatible with 'CompletableFuture<Boolean>' returned from Lightbulb.getLightbulbPowerState() (mismatching null 
	 constraints)
```
l. 5
```
Null type mismatch: required '@NonNull Boolean' but the provided value is null
```

l.5. This is easy, we can use `CompletableFuture<@Nullable Boolean>`, but there is no way to get rid of the other error `@Nullable CompletableFuture<@Nullable Boolean>` is also wrong and the suggested solution is to change the upstream return-type to @NonNull which is obviously not possible.

Reply:
I didn't notice it was because of the overrides. If it is not easily fixable I think it's better to keep the annotations as they are now.